src/verifiable/credential.ts

```diff
@@ -146,6 +271,53 @@ export class W3CCredential {
     }
     return undefined;
   }
+
+  private async validateDIDDocumentAuth(did: DID, resolverURL: string, state: Hash) {
+    const vm = await this.resolveDIDDocumentAuth(did, resolverURL, state);
+    if (!vm) {
+      throw new Error(`can't resolve DID document`);
+    }
+    // published or genesis
+    if (!(vm as any).published && !this.isGenesis(did, state)) {
+      throw new Error(`issuer state not published and not genesis`);
+    }
+  }
+
+  private async resolveDIDDocumentAuth(
```

Comment:
maybe it makes sense to move this separate file, like did-resolver.ts, and make this function public?

Reply:
moved to existed did-helper.ts file